### ISSUE TITLE
ER: Donation components are not small enough to be reused across the site

### ISSUE THREAD

**Issue by ExperimentsInHonesty** [hackforla/website#4407]
### Emergent Requirement - Problem
We have a couple of issues that deal with donation elements across the site and we need a coordinated set of issues for all the changes

### Issue you discovered this emergent requirement in
- #4263 
- #3983 

### Date discovered 
2023-03-26

### Did you have to do something temporarily
- [x] YES I had to make a slide deck to sort it all out
- [ ] NO

### Who was involved
@ExperimentsInHonesty @Thinking-Panda @roslynwythe 

### What happens if this is not addressed
No one knows what to do and the whole thing is stalled.

### Resources
[Donation 2023-03-26 slides](https://docs.google.com/presentation/d/1sD5TAtCe7lQkxFMMbNAHU99Z9DZMhwR_vGV5xFC5-q4/edit) 

### Recommended Action Items
- [ ] Make a new issue
- [ ] Discuss with team
- [ ] Let a Team Lead know

### Potential solutions [draft]
- [ ] Make issues
#### Issue 1
Title:
```
Make donation elements into components that are resuable
```
```
### Overview
We need to make donation elements into components that are reusable so that we can replace the ones that are in use across the site

### Action Items
- [ ] Create or revise issues to make the following includes that could be used in either header or a page card on any page
  - [ ] Circles (issue needs revision https://github.com/hackforla/website/issues/4263)
     - The current image is https://github.com/hackforla/website/blob/gh-pages/assets/images/donate/donation-chart.svg
     - will be replaced with CSS/html that gets put into an include
     - Go to any page that currently uses it, and wherever the image is specified, change it out for the include.  Page should look exactly the same when done.
  - [ ] Donation GIF and Text (not title) (issue needs to be made)
     - is part of an include now.  Take the part minus the header and put it in its own include
       <details>
       <summary>see the part to put in the include</summary>

       ![image](https://user-images.githubusercontent.com/37763229/229888253-5662b326-d705-44bb-8a84-76d6a98ee2c5.png)
      </details>

     - Go to any page or code that currently uses it, and wherever the text and image are specified, change it out for the include.  Pages should look exactly the same when done.
  - [ ] How to Help us Do Impactful work
     - The current html should be moved to an include without the title
        <details>
        <summary>see the part to put in the include</summary>
  
        ![image](https://user-images.githubusercontent.com/37763229/229927209-26ca506f-ee9e-4924-bd9b-229ef1c83e14.png)
        </details>

     - Go to any page or code that currently uses it, and wherever the text is specified, change it out for the include.  Pages should look exactly the same when done.
</details>

- [ ] Review the slide deck, Bonnie made, that shows the changes needed to all the pages where donation elements exist
  - [ ] identify what other issues need to get made and list them.  Get signoff by dev lead and product
     - [ ] Create remaining issues from list

### Resources/Instructions
[Donation 2023-03-26 slides](https://docs.google.com/presentation/d/1sD5TAtCe7lQkxFMMbNAHU99Z9DZMhwR_vGV5xFC5-q4/edit) 
```

See preview of the issue in the comment below

**Comment by ExperimentsInHonesty** [comment]
### Overview
We need to make donation elements into components that are reusable so that we can replace the ones that are in use across the site

### Action Items
- [ ] Create issues to make the following includes that could be used in either header or a page card on any page
  - [ ] Circles
     - The current image is https://github.com/hackforla/website/blob/gh-pages/assets/images/donate/donation-chart.svg
     - will be replaced with CSS/html that gets put into an include
     - Go to any page that currently uses it, and wherever the image is specified, change it out for the include.  Page should look exactly the same when done.
  - [ ] Donation GIF and Text (not title)
     - is part of an include now.  Take the part minus the header and put it in its own include
<details>
<summary>see the part to put in the include</summary>

![image](https://user-images.githubusercontent.com/37763229/229888253-5662b326-d705-44bb-8a84-76d6a98ee2c5.png)
</details>

     - Go to any page that currently uses it, and wherever the image is specified, change it out for the include.  Page should look exactly the same when done.
- [ ] Review the slide deck, Bonnie made, that shows the changes needed to all the pages where donation elements exist
  - [ ] identify what other issues need to get made and list them.  Get signoff by dev lead and product
     - [ ] Create remaining issues from list

### Resources/Instructions
[Donation 2023-03-26 slides](https://docs.google.com/presentation/d/1sD5TAtCe7lQkxFMMbNAHU99Z9DZMhwR_vGV5xFC5-q4/edit)

**Comment by ExperimentsInHonesty** [comment]
# Preview the issue 

### Overview
We need to make donation elements into components that are reusable so that we can replace the ones that are in use across the site

### Action Items
- [x] Create or revise issues to make the following includes that could be used in either header or a page card on any page
  - [x] Circles (issue needs revision https://github.com/hackforla/website/issues/4263)
     - The current image is https://github.com/hackforla/website/blob/gh-pages/assets/images/donate/donation-chart.svg
     - will be replaced with CSS/html that gets put into an include
     - Go to any page that currently uses it, and wherever the image is specified, change it out for the include.  Page should look exactly the same when done.
  - [x] Donation GIF and Text (not title) (issue needs to be made)
     - is part of an include now.  Take the part minus the header and put it in its own include
       <details>
       <summary>see the part to put in the include</summary>

       ![image](https://user-images.githubusercontent.com/37763229/229888253-5662b326-d705-44bb-8a84-76d6a98ee2c5.png)
      </details>

     - Go to any page or code that currently uses it, and wherever the text and image are specified, change it out for the include.  Pages should look exactly the same when done.
  - [x] How to Help us Do Impactful work
     - The current html should be moved to an include without the title
        <details>
        <summary>see the part to put in the include</summary>
  
        ![image](https://user-images.githubusercontent.com/37763229/229927209-26ca506f-ee9e-4924-bd9b-229ef1c83e14.png)
        </details>

     - Go to any page or code that currently uses it, and wherever the text is specified, change it out for the include.  Pages should look exactly the same when done.
</details>

- [x] Review the slide deck, Bonnie made, that shows the changes needed to all the pages where donation elements exist
  - [ ] identify what other issues need to get made and list them.  Get signoff by dev lead and product
     - [ ] Create remaining issues from list

### Resources/Instructions
[Donation 2023-03-26 slides](https://docs.google.com/presentation/d/1sD5TAtCe7lQkxFMMbNAHU99Z9DZMhwR_vGV5xFC5-q4/edit)

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe Please review this ER and the draft of the issue I wrote and see if that is going to solve everything that is referenced in the deck, including the notes at the end.

**Comment by roslynwythe** [comment]
- @ExperimentsInHonesty I renamed #4263 but I need clarification re: mobile vs. desktop views.  In the Action Items above you reference this as the current image:

![image](https://user-images.githubusercontent.com/5314153/229995339-b1492ca4-e492-4fb9-a2c3-990f82c95354.png)

You indicate that the "horizontal" circles layout above should be converted to CSS and referenced in Donate and Join Us.  Is that your intention for both mobile and desktop versions?  I'm confused because in #3983, the newer "vertical" circle image from Figma was referenced.  The instructions indicated we were supposed to extrapolate from the mobile to the desktop design.  The new image was apparently designed to provide larger, more compliant text and I assumed that the new image would be used in both mobile and desktop views.  I apologize if that assumption was incorrect.  I did check the deck but it doesn't really address the adoption of the new circles design.

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty Please see my comment/question above.  This is just a minor additional note: looking at the source for Donate and Join Us pages, it appears that both pages currently reference a different svg file than the one you mentioned: https://github.com/hackforla/website/blob/gh-pages/assets/images/join-us/donation-graphic.svg.  Both images appear the same.  I can't find any references to donation-chart.svg in our site's html.

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe First let me say, I am so grateful for your attention to detail on this issue.  It would be so easy for us to break a few pages without even noticing until it's too late!

I am going to attempt to answer the spirit of your question because this issue has been going on for so long, I want to make sure I don't make mistakes because I am finding it hard to follow every single nuance without writing all the issues myself.  So with that in mind, I am empowering you to make some decisions on your own for this issue with the following guidance

The goals are to 

1. Have a plan for making all these changes so that we can review it and ensure we have not left anything out
2. Make each component modular, so that if something appears on the site in more than one place, we only code it once, and we can reuse it throughout the site.
3. Components should have breakpoints and layouts for both Desktop and Mobile and any other breakpoints we are currently routinely supporting
4. Make issues as small as possible to ensure they are likely not to create scope creep.
5. Make sure we delete any images or other artifacts after they are no longer needed.  We could do this in one cleanup issue, or on each issue as appropriate.  But given that we won't be fixing it all in one big pr, an issue might have to leave an image or include behind that another page is currently using, with a note on the epic or final cleanup issue. 

If you need more support on this issue, I suggest we leave it for Sunday or Monday, where you can demo it to me.  I can also recommend that if you want to talk to someone else on this issue either Jessica or Justin are good candidates for that.

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty Ok, I will proceed with the assumption that we can use the new "vertical" circles design for both mobile and desktop.  If that is a mistake please let me know.

**Comment by roslynwythe** [comment]
### Progress
- For a list of the new issues, please refer to Epic #4492

**Comment by roslynwythe** [comment]
- Created new Epic #4492

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty Above in the "Preview the Issue" section https://github.com/hackforla/website/issues/4407#issuecomment-1496642310 there is an Action Item to make an include from  "How to help us do impactful work".  However in the issue for implementing the new Donate page design you wrote https://github.com/hackforla/website/issues/3983#issuecomment-1484016722 replacing that header and list with a new  header and text ("Provide In-Kind Software & Services").  So will the new "In Kind" include replace the "How to help us do impactful work" include?

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe this is too complicated to talk about async, please add to the top of the dev agenda for Monday.

**Comment by ExperimentsInHonesty** [comment]
- Epic got created https://github.com/hackforla/website/issues/4492
